yubioath-desktop: tries to do git clone on mac

**Describe the bug**
yubioath-desktop tries to do pip install and git clone when building on darwin.

**To Reproduce**
Steps to reproduce the behavior:
1. use darwin machine
2. nix-build -A yubioath-desktop

**Expected behavior**
A reproducible build.

**Screenshots**
https://github.com/NixOS/nixpkgs/pull/104823#issuecomment-733308909

```
  WARNING: The directory '/homeless-shelter/Library/Caches/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
  Collecting git+https://github.com/Yubico/yubikey-manager.git (from -r requirements.txt (line 1))
    Cloning https://github.com/Yubico/yubikey-manager.git to /private/tmp/nix-build-yubioath-desktop-5.0.4.drv-0/pip-req-build-l2b1ssc4
    ERROR: Error [Errno 2] No such file or directory: 'git' while executing command git clone -q https://github.com/Yubico/yubikey-manager.git /private/tmp/nix-build-yubioath-desktop-5.0.4.drv-0/pip-req-build-l2b1ssc4
  ERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?
  make: *** [Makefile:437: pymodules] Error 1
  make: *** Waiting for unfinished jobs....
cannot build derivation '/nix/store/pz6gryhgnm71gycb03vm2i4gp7nr80nc-env.drv': 2 dependencies couldn't be built
[11 built (2 failed), 134 copied (760.0 MiB), 177.6 MiB DL]
error: build of '/nix/store/pz6gryhgnm71gycb03vm2i4gp7nr80nc-env.drv' failed
```

**Notify maintainers**
@Mic92 @risson 

**Metadata**
```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 19.6.0, macOS 10.15.7`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.7`
 - channels(user): `"home-manager, nixpkgs-21.03pre246624.cfed29bfcb2"`
 - nixpkgs: `/Users/user/.cache/nixpkgs-review/pr-104823/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/misc/yubioath-desktop/0001-replace-git-with-normal-python-package.patch
+++ b/pkgs/applications/misc/yubioath-desktop/0001-replace-git-with-normal-python-package.patch
@@ -1,0 +1,24 @@
+From 5eb4d6a384753896d495b09d8651c48672ef234d Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?J=C3=B6rg=20Thalheim?= <joerg@thalheim.io>
+Date: Wed, 25 Nov 2020 11:26:49 +0100
+Subject: [PATCH] replace git with normal python package
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Signed-off-by: Jörg Thalheim <joerg@thalheim.io>
+---
+ requirements.txt | 2 +-
+ 1 file changed, 1 insertion(+), 1 deletion(-)
+
+diff --git a/requirements.txt b/requirements.txt
+index ef25aad..b4246a4 100644
+--- a/requirements.txt
++++ b/requirements.txt
+@@ -1,2 +1,2 @@
+-git+https://github.com/Yubico/yubikey-manager.git
++yubikey-manager
+ ./py/qr
+-- 
+2.29.2
+

--- a/pkgs/applications/misc/yubioath-desktop/default.nix
+++ b/pkgs/applications/misc/yubioath-desktop/default.nix
@@ -19,6 +19,10 @@ mkDerivation rec {
 
   nativeBuildInputs = [ qmake makeWrapper python3.pkgs.wrapPython ];
 
+  patches = [
+    ./0001-replace-git-with-normal-python-package.patch
+  ];
+
   postPatch = ''
     substituteInPlace deployment.pri \
       --replace '/usr/bin' "$out/bin"

